h/auth/util.py

```diff
@@ -54,6 +55,29 @@ def group_principals(user):
     return ['group:{group.pubid}'.format(group=group) for group in user.groups]
 
 
+def has_permission(request, context, userid, permission):
+    """
+    Return True if userid has permission on context object.
+
+    Return True if the given userid has the given permission on the given
+    context object, False otherwise.
+
+    For example:
+
+        if has_permission(request,
+                          some_annotation,
+                          'acct:philip@hypothes.is',
+                          'read'):
+            print 'philip can read this annotation'
+        else:
+            print "philip isn't allowed to read this annotation"
+
+    """
+    policy = request.registry.queryUtility(authorization.IAuthorizationPolicy)
+    users_principals = effective_principals(userid, request)
+    return bool(policy.permits(context, users_principals, permission))
```

Comment:
Agreed that this should be tested. You don't need to mock everything. As you suggested, just set up a basic check that with an `ACLAuthorizationPolicy`, and an appropriate context object, this returns the right values on the basis of stubbed effective principals and the permission name.